### PR TITLE
Add a Quick start section in the README file

### LINKED ISSUE #33: Repository doesn't have a quick start section in the README
**Is your change request related to a problem? Please describe.**
As described in the [documentation section of the Library Guidelines](https://github.com/purescript-contrib/governance/blob/main/library-guidelines.md#documentation), Contributors libraries are expected to have in their README a short summary of the library's purpose, installation instructions, a quick start with a minimal usage example, and links to the documentation and contributing guide.

This library currently doesn't have a quick start in the README.

**Describe the solution you'd like**
The library needs a quick start section after the installation instructions. [argonaut-codecs](https://github.com/purescript-contrib/purescript-argonaut-codecs#quick-start) is one example of a library with a quick start.

**Additional context**
See the [Governance repository](https://github.com/purescript-contrib/governance) for more information about requirements in the Contributors organization.

### PR DESCRIPTION
**Pull request by vladciobanu**
**Description of the change**
FIxes #33 . I added a simple (pure) example using `Ιdentity` over basic operations over integers and strings.

I did not update the changelog since it doesn't affect the code in any way. Should I?

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation in the README and/or documentation directory
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by paluh** [comment]
@vladciobanu Thanks a lot for this contribution! This is really cool!

The below is rather a side note which I want to discuss. I wasn't sure where to put these thoughts so I'm writing it here :-)

I wonder if we can add a direct reference to the Mealy definition (copy it from the wikipedia or something) somewhere here and show how it corresponds to the type `MealyT f s a`. I want to do this because from my perspective it is a quite interesting twist in this implementation that `state` is not directly represented in the type but we have only input (`s`) and output (`a`) and context (`f`) types.
In other words I propose some examples which show how to carry state using probably plain closure. This is a copy from Haskell machines lib which clearly separates state from the input:

```haskell
>>> let countingMealy = unfoldMealy (\i x -> ((i, x), i + 1)) 0
>>> run (auto countingMealy <~ source "word")
[(0,'w'),(1,'o'),(2,'r'),(3,'d')]
```

and here we have an implementation which uses closure:

```haskell
-- | A 'Mealy' machine modeled with explicit state.
unfoldMealy :: (s -> a -> (b, s)) -> s -> Mealy a b
unfoldMealy f = go where
  go s = Mealy $ \a -> case f s a of
    (b, t) -> (b, go t)
```

Of course the above snippets are related to the simpler representation of Mealy:

```haskell
newtype Mealy a b = Mealy { runMealy :: a -> (b, Mealy a b) }
```

In our case we can probably use this but also just provide an example using `StateT / State`. I can add this in a separate PR. What do you think?

**Comment by thomashoneyman** [comment]
> I wonder if we can add a direct reference to the Mealy definition (copy it from the wikipedia or something) somewhere here and show how it corresponds to the type MealyT f s a. I want to do this because from my perspective it is a quite interesting twist in this implementation that state is not represented directly in the type but we have only input (s) and output (a) and context (f) params.

I think this would be a nice addition.

> I propose to add some examples which show how to carry state using just plain closure but also by using State.

I think this is also a nice idea. We should endeavor to keep the quick start small (just enough to give you the context and most basic examples to get you started with the library), so as long as we can capture this in a quick example or two I think it makes a great fit.

There's also the `docs/` folder for any longer writeups or treatments of this! For example, that might be a good place for a longer explanation of what Mealy machines are, why they're useful, and how they correspond to `MealyT`.

> and here we have an implementation of unfoldMealy which should probably be added to the lib (?):

Also a nice idea.

@vladciobanu @paluh I don't think these additions need to hold up this PR, and could be done in a subsequent one if you'd prefer to handle them @paluh -- unless you'd rather do it, @vladciobanu. We can iterate on these docs sections :)

File: README.md
Comment (thomashoneyman) [inline README.md:24]: I think it would be nice to have a one-line description of the `MealyT` type before mentioning `Identity`. For example:

```suggestion
Mealy machines are finite state machines. The `MealyT f s a` type represents a machine where `f` is the effect used for evaluation, `s` is the input state, and `a` is the output value. The examples here use `Identity` as the effect type for simplicity, but you would usually use a different `Monad` such as `Effect`, `Aff`, or `State`.
```

File: README.md
Comment (thomashoneyman) [inline README.md:44]: We usually represent this with a pseudo-repl session:

---

The resulting machine can be materialized by transforming it into an array:

```purs
> toUnfoldable unit machine1 :: Array String
["5","10","15","20","25"]
```

---

though it's not necessary to do so. Just if you'd like!

File: README.md
Comment (thomashoneyman) [inline README.md:61]: For parity with the earlier example, should these two be merged?

```purs
    scale :: Int -> Step Identity Int Int
    scale n
      | n == 0 = Halt
      | otherwise = Emit (n `div` 2) $ pureMealy scale
```

File: README.md
Comment (thomashoneyman) [inline README.md:46]: It may also be useful to note here that we're composing `Step`s and briefly note what a `Step` is. Or perhaps it would be better to note that in the prior section. Either way, steps are fundamental to mealy machines and I think we should add a quick sentence about them in the quick start.

Perhaps:

> A `Step f s a` represents a state transition in the machine. When you run a machine you are executing a series of steps. At each step the machine can stop via the `Halt` constructor or emit a value and construct the rest of the machine.

Though feel free to phrase however you see fit.

File: README.md
Comment (paluh) [reply README.md:24]: I'm not sure if I'm right but I think that `s` should be rather called an `input`. If my understanding of `MealyT` type is correct I'm opting for changing parameters names to `MealyT f i o` too.
P.S.
A somewhat interesting topic for a broader discussion from my perspective is: how this type really differs from the `Star f i o` from profunctors... ;-)

Comment (thomashoneyman) [reply README.md:24]: `i` and `o` make more sense to me as well, but I'm not familiar with why `s` was originally chosen. I am personally fine with changing it and just referring to a machine's input and output throughout documentation comments and the quick start, but I don't have a strong opinion on it.

File: README.md
Comment (thomashoneyman) [inline README.md:42]:
```suggestion
scales the inputs (by dividing by 2). The result is then transformed into a string.
```

**Comment by paluh** [comment]
@thomashoneyman Should I move the above comment to some separate issue like "Document correspondence of `MealyT` to the abstract model" or something?

**Comment by thomashoneyman** [comment]
If it's something to go in the quick start then I'd like to let @vladciobanu have a chance to see whether and if he'd like to work that in to his existing content here, but if it's for the `docs` directory then feel free to go ahead with the issue and I can assign it to you!

File: README.md
Comment (vladciobanu) [reply README.md:24]: I think it might be nicer to rename these to `input` and `output`, but I would do it in a subsequent PR because it would be out of place to change the module and module docs in this PR.

File: README.md
Comment (vladciobanu) [reply README.md:76]: You are right! I changed the examples a bit, using `guard` in the monadic version. Does this look better?

File: README.md
Comment (vladciobanu) [reply README.md]: I agree with this! We are composing machines, not steps, but explaining what a step is is quite useful!

**Comment by vladciobanu** [comment]
I think the wikipedia definition would be a great addition to the `Mealy` module documentation rather than the Quick start section, or even the README.md file. I say this mostly because I assume we might add more machines to this library. I'm happy to take this on in a subsequent PR!

I also think that more in-depth examples belong in either tests or docs, again, in subsequent PR's.

I also added imports to the examples. Mostly because I decided to use `guard` and that's not in `Prelude`, so I figured it's going to be easier to run the examples.

File: README.md
Comment (thomashoneyman) [reply README.md:76]: If you are comfortable switching to two spaces instead of four in the example (as is done in the `do` example above) I'd appreciate that, but otherwise 👍!

**Review by thomashoneyman — APPROVED** [review]
Thanks for adding this, @vladciobanu. I think that we should create followup issues or get right into the PRs for the changes that you and @paluh mentioned in this thread. We may need to briefly chat about who is going to do what just to make sure that there isn't any wasted / overlapping work. We could do that in this PR thread or in a chat on Discourse -- whatever y'all prefer :)

File: README.md
Comment (thomashoneyman) [reply README.md]: 👍 sounds good to me.

File: README.md
Comment (thomashoneyman) [reply README.md:76]: I'm going to leave this unmerged to give you a moment to respond to this, but at that point I will go ahead and merge.

**Comment by thomashoneyman** [comment]
Also, once this is merged I will release a new version of the library and push the docs to Pursuit. Then we'll have a nice landing page for `-machines` and some module docs!

**Comment by vladciobanu** [comment]
I propose we create a few issues:

- Explain Mealy in Module Docs: Add wikipedia definition and explain how it connects to the MealyT type.
- Rename Type Variables: Rename `MealyT f s a` to `MealyT effect input output` or `MealyT eff i o`

I also propose we add the details that @paluh wrote in his original comment here to the docs issue, as an example of something that would be useful to be added there.

**Comment by thomashoneyman** [comment]
Sounds good to me! Let’s start there.

**Pull request merged**
